### PR TITLE
LPS-125550 log4j2

### PR DESCRIPTION
**Pull request by yuhai**
cc @dantewang, @tinatian

### CODE REVIEW

**Comment by yuhai** [comment]
ci:test:sf

**Comment by yuhai** [comment]
ci:test:relevant

File: portal-impl/build.xml [hunk 1]
```diff
@@ -539,7 +539,7 @@ svn://svn.liferay.com/repos/public/alloy/trunk/sandbox/taglibs.
 				/>
 
 				<copy
-					file="classes/log4j.properties"
+					file="classes/log4j2.properties"
```
Comment (tinatian) [inline portal-impl/build.xml:542]: @yuhai @dantewang Is there any reason we have to rename?

Tina.

File: build-common.xml [hunk 2]
```diff
@@ -1850,7 +1850,7 @@ artifact.url=${remote.repository.cdn.url}/com/liferay/portal/com.liferay.${proce
 				<path refid="lib.pre.classpath" />
 				<fileset
 					dir="${project.dir}/lib/development"
-					excludes="osgi-annotation-versioning.jar,jalopy.jar,jetty*.jar,tomcat*.jar"
+					excludes="jalopy.jar,jetty*.jar,tomcat*.jar"
```
Comment (tinatian) [inline build-common.xml:1853]: @dantewang @yuhai Why this is causing unit test compile faile? why we did not see it on log4j1?

File: portal-test/src/com/liferay/portal/test/log/CaptureAppender.java [hunk 3]
```diff
@@ -14,57 +14,80 @@
...
-		_logger.setLevel(_level);
+		String loggerName = _logger.getName();
+		String loggerConfigName = _loggerConfig.getName();
 
-		try {
-			_parentField.set(_logger, _parentCategory);
-		}
-		catch (Exception exception) {
-			throw new RuntimeException(exception);
+		if (!loggerName.equals(loggerConfigName)) {
```
Comment (tinatian) [inline portal-test/src/com/liferay/portal/test/log/CaptureAppender.java:82]: @yuhai @dantewang  When will the logger name is not same as logger config?

File: modules/apps/petra/petra-log4j/src/main/java/com/liferay/petra/log4j/internal/CentralizedConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+		// Filter is defined. Since Filters are not named duplicates may be
+		// present.
+
+		Filter newFilter = abstractConfiguration.getFilter();
+
+		if (newFilter != null) {
+			newFilter.start();
+		}
+
+		addFilter(newFilter);
```
Comment (tinatian) [inline modules/apps/petra/petra-log4j/src/main/java/com/liferay/petra/log4j/internal/CentralizedConfiguration.java:134]: @dantewang Please move this line to the if block above.

Tina.

File: modules/apps/petra/petra-log4j/src/main/java/com/liferay/petra/log4j/internal/CentralizedConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+				continue;
+			}
+
+			addLogger(name, newLoggerConfig);
+
+			newLoggerConfig.start();
+		}
+	}
+
+	private void _updateLoggers() {
```
Comment (tinatian) [inline modules/apps/petra/petra-log4j/src/main/java/com/liferay/petra/log4j/internal/CentralizedConfiguration.java:233]: @dantewang The logic here is simple, please inline this method.

Tina.

**Pull request closed without merging**